tools/gulptasks/dist-compress.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) Highsoft AS
+ */
+
+const gulp = require('gulp');
+const fs = require('fs-extra');
+const zlib = require('zlib');
+const glob = require('glob');
+const log = require('./lib/log');
+const build = require('./lib/build');
+const yargs = require('yargs');
+const zip = require('gulp-zip');
+const stream = require('stream');
+const util = require('util');
+
+
+const DIST_DIR = 'build/dist';
+
+
+/**
+ * Creates zip files for highcharts, highstock, highmaps, highcharts-gantt.
+ *
+ * @return {Promise<any> | Promise | Promise} Promise to keep
+ */
+function distZip() {
+    const properties = build.getBuildProperties();
+
+    const zipTasks = Object.keys(properties).map(key => {
+
+        const { product } = properties[key];
+        const { name, version } = product;
+
+        const dirToZip = `${DIST_DIR}/${key}`;
+        if (!fs.existsSync(dirToZip)) {
+            return Promise.reject(new Error(`Missing folder: ${dirToZip}. Has the other dist tasks been run in advance?`));
+        }
+
+        const zipFileName = `${name.replace(/ /g, '-')}-${version}.zip`;
+        const excludedDir = `!${dirToZip}/js-gzip/**`;
+        log.message(`Zipping file: ${zipFileName}, excluding sub-directory ${excludedDir}`);
+
+        return new Promise((resolve, reject) => {
+            gulp.src([dirToZip + '/**', excludedDir])
+                .pipe(zip(zipFileName))
+                .pipe(gulp.dest(DIST_DIR))
+                .on('error', reject)
+                .on('end', resolve);
+        });
+    });
+
+    return Promise.all(zipTasks);
+}
+
+/**
+ * Creates gzipped version of files in build/dist/{productname}/code and stores them
+ * in build/dist/{productname}/js-gzip. E.g. build/dist/highcharts/code --> gzip and store in build/dist/highcharts/code/js-gzip
+ *
+ * @return {Promise<any> | Promise | Promise} Promise to keep
```

Comment:
I do not understand the type here. If it gets always solved to the last `then` without returning something inside `then`, it should be a simple `Promise<void>`.

Reply:
Fixed by removing the last then() as it is not really needed. And changed return type to Promise<*> for now.